framework/source/class/qx/ui/window/Window.js

```diff
@@ -993,6 +1018,112 @@ qx.Class.define("qx.ui.window.Window",
       }
     },
 
+    // overridden
+    _transformCenterWhen : function(value)
+    {
+      var             values = value;
+      
+      // Validate type of value
+      if (typeof value != "string" && ! (value instanceof Array))
+      {
+        throw new Error(
+          "centerWhen requires a string or an array of strings");
+      }
+
+      // Convert the value to an array if it was given as a string.
+      if (typeof value == "string")
+      {
+        if (value.trim() === "")
+        {
+          // We were given an empty string or array, requesting no automatic
+          // centering. Create an empty array of centering behaviors.
+          values = [];
+        }
+        else
+        {
+          // We were given a string. If they put multiple values into the
+          // string, separate them out into separate values, and place all into
+          // an array.
+          values = qx.lang.String.clean(value).split(/\s+/);
+        }
+      }
+
+      // Validate
+      values.forEach(
+        function(value)
+        {
+          switch(value)
+          {
+          case "appear" :
+          case "resize" :
+            // These are expected values
+            break;
+
+          default :
+            throw new Error("Unexpected centerWhen value: " + value);
+          }
+        });
+
+      return values;
+    },
```

Comment:
this transform is perfectly valid code, but it seems very complicated for what the end goal is trying to achieve, ie a simple pair of booleans.  How about changing the property to `check: [ "appear", "resize", "both", "never" ]`?  It's a common approach to this problem and a lot easier to maintain